modules/core/src/test/java/org/apache/ignite/internal/ThreadNameValidationTest.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.binary.BinaryMarshaller;
+import org.apache.ignite.internal.processors.cache.GridCacheAbstractSelfTest;
+import org.apache.ignite.testframework.junits.GridAbstractTest;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runners.model.Statement;
+
+/**
+ * Check threads for default names in single and thread pool instances.
+ */
+public class ThreadNameValidationTest extends GridCacheAbstractSelfTest {
+
+    /** {@link Executors.DefaultThreadFactory} count before test. */
+    private static transient int defaultThreadFactoryCountBeforeTest;
+
+    /** {@link Thread#threadInitNumber} count before test. */
+    private static transient int anonymousThreadCountBeforeTest;
+
+
+    /** */
+    private static final TestRule beforeAllTestRule = (base, description) -> new Statement() {
+        @Override public void evaluate() throws Throwable {
+            defaultThreadFactoryCountBeforeTest = getDefaultPoolCount();
+            base.evaluate();
+        }
+    };
+
+    /** Manages before first test execution. */
+    @ClassRule public static transient RuleChain firstLastTestRule
+        = RuleChain.outerRule(beforeAllTestRule).around(GridAbstractTest.firstLastTestRule);
+
+    /** */
+    private static final String FIELD = "user-name";
+
+    /** */
+    private final ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
+
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        anonymousThreadCountBeforeTest = getAnonymousThreadCount();
+        super.beforeTest();
+    }
+
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        assertEquals("Executors.DefaultThreadFactory usage detected, IgniteThreadPoolExecutor is preferred",
+            defaultThreadFactoryCountBeforeTest, getDefaultPoolCount());
+        assertEquals("Thread without specific name detected",
+            anonymousThreadCountBeforeTest, getAnonymousThreadCount());
+        super.afterTest();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected int gridCount() {
+        return 1;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected CacheConfiguration cacheConfiguration(String igniteInstanceName) throws Exception {
+        CacheConfiguration ccfg = super.cacheConfiguration(igniteInstanceName);
+
+        ccfg.setStoreKeepBinary(true);
+
+        return ccfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        cfg.setMarshaller(new BinaryMarshaller());
+
+        return cfg;
+    }
```

Comment:
It's better to use node with persistence.
```
cfg.setDataStorageConfiguration(
            new DataStorageConfiguration()
                .setDefaultDataRegionConfiguration(
                    new DataRegionConfiguration()
                        .setPersistenceEnabled(true)
                )
        );
```

Reply:
Unfortunately, persistence uses AsyncFileIO, which depends on FileSystemProvider, and its not clear which side effects we will achieve with specific thread pool.

Reply:
You can configure fileIOFactory manually. You may use - new DataStorageConfiguration().setFileIOFactory(*) or you can set system property IGNITE_USE_ASYNC_FILE_IO_FACTORY to false. See org.apache.ignite.configuration.DataStorageConfiguration#fileIOFactory